### DISCUSSION TITLE
Set up Keycloak behind reverse Proxy - Steps - Please review

### DISCUSSION THREAD

**Discussion by kunalgithub** [keycloak/keycloak#30906] [Q&A]
Domain:     `qa25-iam.acme.com`
Webserver: Apache 2.4.X 
Should be available on : `https://qa25-iam.acme.com/auth`

Changes done

Parameters to Keycloak 

1. --http-relative-path=auth
2.  --proxy-headers=xforwarded

```
/acme/apps/keycloak-25.0.1/bin/kc.sh --verbose start-dev --http-port 7777 --log="file" --log-file="/acme/logs/keycloak-25.0.1-logs/keycloak.log" --proxy-headers=xforwarded --http-relative-path=auth
```

Changes done in keycloak.conf

```
hostname=https://qa25-iam.acme.com/auth
hostname-path=/auth
```

Apache Conf - /etc/apache2/sites-enables/ssl-qa25-iam.acme.com.conf **As per your naming convention**

```
ProxyPass /auth http://0.0.0.0:7777/auth
ProxyPassReverse /auth http://0.0.0.0:7777/auth
```

**Do suggest these steps are fine for documentation**

**Comment by codespearhead** [comment]
1. `start-dev` [shouldn't be used in production](https://github.com/keycloak/keycloak/discussions/28279#discussioncomment-9001818).
2. `hostname-path` [is deprecated](https://www.keycloak.org/server/reverseproxy#:~:text=Relevant%20options). In your case [you can just get rid of it](https://github.com/keycloak/keycloak/blob/a8aa410ad350f550bc60e82f4beda32eb064528f/docs/documentation/upgrading/topics/changes/changes-25_0_0.adoc#new-hostname-options). You can also set it via an [environment variable or a CLI flag btw](https://www.keycloak.org/server/all-config?q=hostname#:~:text=Hostname%20v2).
3. Change `0.0.0.0` to `127.0.0.1` so that the Keycloak server isn't directly accessible via intranet. If you can run it in a separate, isolated machine with a static IP, even better.

Related: https://github.com/keycloak/keycloak/discussions/28279
